### PR TITLE
Gateway UI Improvements

### PR DESCRIPTION
**Pull request by Kodylow**
## Summary by CodeRabbit

- **New Features**
	- Introduced `Error`, `ErrorMessage`, `WalletCard`, `FederationsTable`, and `Loading` components to enhance user interface functionality.
	- Added a new `rotate` command for improved process management.

- **Improvements**
	- Enhanced localization for wallet-related interfaces, providing clearer labels and instructions.
	- Improved layout for `FederationCard` and added wallet information display.
	- Transitioned `ConnectFederation` to a modal interface for better user experience.

- **Bug Fixes**
	- Rectified clarity issues in user guidance for shutdown procedures.

- **Documentation**
	- Updated language files to improve user understanding and interaction.

### CODE REVIEW

File: apps/gateway-ui/src/components/wallet/WalletCard.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  balanceMsat,
+  network,
+}: WalletCardProps): JSX.Element {
+  const { t } = useTranslation();
+  const { gateway } = React.useContext(ApiContext);
+
+  const [tabIndex, setTabIndex] = useState(0);
+  const [amount, setAmount] = useState<number>(0);
+  const [address, setAddress] = useState<string>('');
+  const [bip21Uri, setBip21Uri] = useState<string>('');
```
[the remark this answers is not in the document]
Comment (alexlwn123) [reply apps/gateway-ui/src/components/wallet/WalletCard.tsx:50]: ^^

File: apps/gateway-ui/src/components/Error.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Flex, Heading, Text, useTheme } from '@chakra-ui/react';
+import { useTranslation } from '@fedimint/utils';
+
+interface ErrorProps {
+  error: string;
+}
+
+export const Error: React.FC<ErrorProps> = ({ error }) => {
```
[the remark this answers is not in the document]
Comment (alexlwn123) [reply apps/gateway-ui/src/components/Error.tsx:9]: ^^

File: apps/gateway-ui/src/App.tsx [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
 import { GatewayInfo, FederationInfo } from '@fedimint/types';
-import { FederationCard, ConnectFederation } from './components';
+import { ConnectFederation, InfoCard } from './components';
 import { GatewayApi } from './GatewayApi';
 import { ApiProvider } from './ApiProvider';
 import { Wrapper, Login } from '@fedimint/ui';
 import { useTranslation } from '@fedimint/utils';
+import { FederationsTable } from './components/federations/FederationsTable';
+import { Loading } from './components/Loading';
+import { Error } from './components/Error';
```
[the remark this answers is not in the document]
Comment (alexlwn123) [reply apps/gateway-ui/src/App.tsx:19]: ^^

File: apps/guardian-ui/src/languages/en.json [hunk 4]
```diff
@@ -149,7 +149,7 @@
         "current-session": "Current Session",
         "session-to-shutdown": "Session to Shutdown at",
         "confirm-shutdown": "Confirm Shutdown",
-        "session-to-shutdown-helper": "Enter the session height at which you want to shutdown your guardian node."
+        "session-to-shutdown-helper": "."
```
[the remark this answers is not in the document]
Comment (alexlwn123) [reply apps/guardian-ui/src/languages/en.json:161]: ^^

File: packages/utils/src/index.tsx [hunk 5]
```diff
@@ -1,3 +1,20 @@
...
+  switch (network) {
+    case Network.Signet:
+      return new URL(`https://mutinynet.com/lightning/node/${nodeId}`);
+    case Network.Testnet:
+      return new URL(`https://mempool.space/testnet/lightning/node/${nodeId}`);
+    case Network.Bitcoin:
+    case Network.Regtest:
+      return new URL(`https://mempool.space/lightning/node/${nodeId}`);
+  }
+};
```
[the remark this answers is not in the document]
Comment (alexlwn123) [reply packages/utils/src/index.tsx:20]: ^^

**Review by alexlwn123 — COMMENTED** [review]
Mostly nits... resolved bad code rabbit suggestions. The remaining ones should probably be addressed. Otherwise, lgtm

File: apps/gateway-ui/src/App.tsx [hunk 6]
```diff
@@ -79,118 +81,91 @@ export const App = React.memo(function Admin(): JSX.Element {
...
-          <Heading size='lg' marginBottom='4'>
-            {t('common.error')}
-          </Heading>
-          <Text fontSize='md'>{error}</Text>
-        </Flex>
-      );
-    }
+  const handleWithdraw = (federation: FederationInfo) => {
+    // Implement withdraw logic here
+    console.log('Withdraw from federation:', federation.federation_id);
```
[the remark this answers is not in the document]
Comment (tvolk131) [reply apps/gateway-ui/src/App.tsx:91]: Maybe we should call `alert()` to let the user know that the button click has been acknowledged

Also, change the comment to `// TODO: Implement withdraw logic here`

File: apps/gateway-ui/src/App.tsx [hunk 7]
```diff
@@ -79,118 +81,91 @@ export const App = React.memo(function Admin(): JSX.Element {
...
-            >
-              {t('admin.fetch-info-modal-text')}
-            </CircularProgressLabel>
-          </CircularProgress>
-        </Flex>
-      );
-    }
+  const handleDeposit = (federation: FederationInfo) => {
+    // Implement deposit logic here
+    console.log('Deposit for federation:', federation.federation_id);
```
[the remark this answers is not in the document]
Comment (tvolk131) [reply apps/gateway-ui/src/App.tsx:86]: Maybe we should call `alert()` to let the user know that the button click has been acknowledged

Also, change the comment to `// TODO: Implement deposit logic here`

**Review by tvolk131 — CHANGES_REQUESTED** [review]

**Comment by Kodylow** [comment]
This was just spitballing on a new UI for the gateway, converting to draft until actually ready for review

**Comment by Kodylow** [comment]
closing in favor of https://github.com/fedimint/ui/pull/489#pullrequestreview-2234052482 , reviewable state

**Comment by tvolk131** [comment]
^

**Pull request closed without merging**